library/std/src/sys/sgx/abi/usercalls/alloc.rs

```diff
@@ -89,9 +89,13 @@ pub unsafe trait UserSafe {
     /// * the pointed-to range is not in user memory.
     unsafe fn from_raw_sized(ptr: *mut u8, size: usize) -> NonNull<Self> {
         assert!(ptr.wrapping_add(size) >= ptr);
-        let ret = Self::from_raw_sized_unchecked(ptr, size);
-        Self::check_ptr(ret);
-        NonNull::new_unchecked(ret as _)
+        // SAFETY: The pointer must be a valid pointer for the entire size.
```

Comment:
This should say something like "The caller has guaranteed that the pointer is valid" -- our intent with these safety comments is basically to enumerate the obligations imposed on us by the function being called (i.e., the stuff in it's safety comment) and for each to explain why it is met.

Reply:
Fixed in latest commit.